benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/tasks/running/ValidateTerminationCriteria.java

```diff
@@ -27,13 +28,31 @@ public ValidateTerminationCriteria(String testID) {
   }
 
   @Override
-  public Boolean call() throws Exception {
+  public TerminationCriteriaResult call() throws Exception {
 
     logger.info("running: " + testID);
 
     CompleteSelectionStrategy completeSelectionStrategy =
         (CompleteSelectionStrategy) explorationModelDAO.getExperimentSelectionStrategy(testID);
 
-    return completeSelectionStrategy.isTestComplete(testID);
+    // TODO - check if can reach goal
+    boolean canReachGoal = true;
+
+    // has regression model
+    boolean hasRegressionModel = explorationModelDAO.hasRegressionModel(testID);
+
+    // TODO - handle cases with regression model
+
+
+    if (completeSelectionStrategy.isTestComplete(testID)) {
+      return TerminationCriteriaResult.GOAL_NO_REGRESSION_EXECUTED;
+    }
+
+    return TerminationCriteriaResult.GOAL_NO_REGRESSION_REMAINING;
+
+  }
+
+  public enum TerminationCriteriaResult {
```

Comment:
Done in https://github.com/benchflow/benchflow/pull/394/commits/51f1a8aefa04cacbcc20128cb82fe45829144d93